source/common/event/scaled_range_timer.h

```diff
@@ -0,0 +1,69 @@
+#include <chrono>
+
+#include "envoy/event/timer.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Event {
+
+class ScaledRangeTimerManager {
```

Comment:
My plan is to have the ThreadLocalOverloadStateImpl construct instances of this class if the to-be-added action is configured. I don't know that we need an interface, but it might be useful for unit testing of the OverloadManagerImpl. Thoughts?